libs/constrained-generators/src/Constrained/Env.hs

```diff
@@ -25,28 +24,21 @@ data EnvValue where
 deriving instance Show EnvValue
 
 data EnvKey where
-  EnvKey :: Typeable a => !(Var a) -> EnvKey
+  EnvKey :: !(Var a) -> EnvKey
```

Comment:
Removing of the Typeable constraint is surprising. At first I thought that the Var constructor had the Typeable constraint, so that made it not required, But I see that is not the case. We must enforce this some where.

Is singletonEnv the only place we make Env?

Reply:
Basically the same reason as before, we have the Var index invariant so we can get away with not doing super expensive type casting